pandas/core/reshape/merge.py

```diff
@@ -774,6 +776,28 @@ def _get_join_info(self):
             join_index = join_index.astype(object)
         return join_index, left_indexer, right_indexer
 
+    def _create_join_index(self, indexer, using_left=True):
+        if using_left:
+            index = self.left.index
+            other_index = self.right.index
```

Comment:
maybe to make this more clear, pass ``how=, index, other_index`` as well (then you don't need the switch